src/shared/UniversalRender.jsx

```diff
@@ -64,15 +69,23 @@ const scrollTop = (el, topOffset, prevDocumentInfo, triesRemaining) => {
     const documentInfo = {
         scrollHeight: document.body.scrollHeight,
         scrollTop: Math.ceil(document.scrollingElement.scrollTop),
-        scrollTarget: calcOffsetRoot(el) + topOffset
+        scrollTarget: calcOffsetRoot(el) + topOffset,
+        direction: prevDocumentInfo.direction,
     };
-
-    if(prevDocumentInfo.scrollTop > (documentInfo.scrollTop + SCROLL_UP_FUDGE_PIXELS)) { //detecting that the user has scrolled in an up direction
-        return;
+    let doScroll = false;
+    if(SCROLL_DIRECTION_DOWN === prevDocumentInfo.direction) {
+        doScroll = (!(prevDocumentInfo.scrollTop > (documentInfo.scrollTop + SCROLL_FUDGE_PIXELS))
```

Comment:
the NOT operator here is confusing, just flip the comparison operator

Reply:
I'll do it, though I dislike  `<=`